Domain Refactors

ChrisCostanzo-acme opened the pull request:
Current tentative changes:

- 'Input' renamed to 'DomainInput'
- 'Output' renamed to 'DomainModel'
- Naming convention for individual outputs changed to: [Feature]DomainTo[Dest]Model. Ex: HomeDomainToUIModel instead of 'HomeUIOutput'.
- 'Success' renamed to 'SuccessUseCaseInput'
- 'Failure' renamed to 'FailureUseCaseInput'
- 'OutputTransformer' renamed to 'DomainModelTransformer'
- 'InputTransformer' renamed to 'DomainInputTransformer'
- Deprecated old classes.
- Updated documentation.
- Upgraded other clean framework packages from clean framework legacy to newest.

ChrisCostanzo-acme committed:
Implement some tentative name changes.

- 'Entity' renamed to 'UseCaseState'
- 'Input' renamed to 'DomainInput'
- 'Output' renamed to 'DomainOutput'
- Naming convention for individual outputs changed to: [Feature]DomainTo[Dest]Output. Ex: HomeDomainToUIOutput
- 'Success' renamed to 'SuccessUseCaseInput'
- 'Failure' renamed to 'FailureUseCaseInput'

ChrisCostanzo-acme committed:
Issue fixes, example refactors.

- Refactored 'form' feature on example.

ChrisCostanzo-acme committed:
Add old classes marked as deprecated.

ChrisCostanzo-acme committed:
Refactor.

ChrisCostanzo-acme committed:
Undo rename of 'Entity' to 'UseCaseState'

- In Clean Architecture, 'Entity' is the name used, so we won't change it to stay within clean architecture terms.

ChrisCostanzo-acme committed:
Continue restore of 'Entity' name.

ChrisCostanzo-acme committed:
Update inputs, outputs.

- Renamed 'DomainOutput' to 'DomainModel'

ChrisCostanzo-acme committed:
Refactors to example project.

ChrisCostanzo-acme committed:
More example project refactors.

ChrisCostanzo-acme committed:
Update form_use_case.dart

ChrisCostanzo-acme committed:
Add basic documentation to data classes.

ChrisCostanzo-acme committed:
Cont'd refactors and documentation.

ChrisCostanzo-acme committed:
Update profile_use_case.dart

ChrisCostanzo-acme committed:
Add deprecations for input/output transformers.

ChrisCostanzo-acme committed:
Update domain_input.dart

ChrisCostanzo-acme committed:
Add updated domain layer documentation.

- Reflects changes to data class names.

ChrisCostanzo-acme committed:
Update domain-layer.mdx

ChrisCostanzo-acme committed:
Update UI layer docs to reflect latest changes.

ChrisCostanzo-acme committed:
Update project structure docs.

ChrisCostanzo-acme committed:
Update intro docs.

ChrisCostanzo-acme committed:
Remove documentation comment.

ChrisCostanzo-acme committed:
Update domain-layer.mdx

ChrisCostanzo-acme committed:
Reorganize example project.

- Domain inputs should be kept in a new file: feature/domain/feature_domain_inputs.dart

ChrisCostanzo-acme committed:
Added new base class for data transfer objects.

- 'DomainInput' and 'DomainModel' now both extend from 'DomainDTO'.

ChrisCostanzo-acme committed:
Rename presenter output/model methods.

ChrisCostanzo-acme committed:
Update presenter.dart

sarbagya-acme commented on packages/clean_framework/example/lib/features/form/presentation/form_presenter.dart:
```diff
@@ -20,7 +20,7 @@ class FormPresenter
   FormViewModel createViewModel(
     FormUseCase useCase,
-    FormUIOutput output,
+    FormDomainToUIModel output,
```
should the output here be renamed as well ?

ChrisCostanzo-acme committed:
Update gateway method params to match new class names, example projects.

- Updated class/param names in clean_framework example and clean_framework_http example.

ChrisCostanzo-acme committed:
More file renames.

ChrisCostanzo-acme committed:
Update domain layer docs to explain use case bridges.

ChrisCostanzo-acme committed:
Update poke_router.dart

ChrisCostanzo-acme committed:
Added theme_example project and started documentation.

- theme_example showcases one way to update a global app theme from within a single feature.
- Added WIP docs explaining this process, not yet complete.
- Updated domain layer docs.

ChrisCostanzo-acme committed:
Finish theme_example documentation.

ChrisCostanzo-acme committed:
Update codelabs config.

ChrisCostanzo-acme committed:
Updated changelog with new version.

ChrisCostanzo-acme committed:
Update clean_framework_http changelog.

ChrisCostanzo-acme committed:
Update CHANGELOG.md

ChrisCostanzo-acme committed:
Add data-driven fixes configuration.

- Running dart fix will replace all deprecated classes with the new ones.
- Does not include parameter or method renames.

ChrisCostanzo-acme committed:
Update CHANGELOG.md

ChrisCostanzo-acme committed:
Update package pubspecs and changelogs.

- Updated pubspecs/changelogs in preparation for publish.

ChrisCostanzo-acme committed:
Rewrote the intro guide to be more thorough and easy to understand.

ChrisCostanzo-acme committed:
Update architecture-overview.mdx

ChrisCostanzo-acme committed:
Docs refactors, change pubspec to point to git until publish.

ChrisCostanzo-acme committed:
Added a rough draft of an article comparing clean framework and other patterns.

- Needs more info.

ChrisCostanzo-acme committed:
Minor docs refactors.

ChrisCostanzo-acme committed:
Add missing info to global theme documentation.

sarbagya-acme commented on packages/clean_framework/pubspec.yaml:
```diff
@@ -1,8 +1,9 @@
 name: clean_framework
 description: Clean Architecture components library, inspired on the guidelines created by Uncle Bob.
-version: 3.2.5
+version: 4.0.0
```
let's publish as a dev version first.
4.0.0-dev.0

sarbagya-acme commented on packages/clean_framework/pubspec.yaml:
```diff
@@ -1,8 +1,9 @@
+version: 4.0.0
 homepage: https://acmesoftware.com/
 repository: https://github.com/AcmeSoftwareLLC/clean_framework
+publish_to: none
```
please remove

sarbagya-acme commented on packages/clean_framework_test/pubspec.yaml:
```diff
@@ -1,16 +1,29 @@
 name: clean_framework_test
 description: A collection test helpers to support clean_framework.
-version: 0.5.2
+version: 0.5.3
```
0.5.3-dev.0

sarbagya-acme commented on packages/clean_framework_test/pubspec.yaml:
```diff
@@ -1,16 +1,29 @@
+version: 0.5.3
 homepage: https://acmesoftware.com/
 repository: https://github.com/AcmeSoftwareLLC/clean_framework/packages/clean_framework_test
+publish_to: none
```
please revert

ChrisCostanzo-acme committed:
Add partial fix/upgrade for clean_framework_graphql.

ChrisCostanzo-acme committed:
Added basic GraphQL example to clean_framework_graphql.

- Later, will duplicate the full clean_framework example app and use GraphQL, so there are examples of both.

ChrisCostanzo-acme committed:
Update pubspecs to be dev versions.

ChrisCostanzo-acme committed:
Add links to example projects to UI layer docs.

ChrisCostanzo-acme committed:
Add more info on examples to docs.

ChrisCostanzo-acme committed:
Update architecture-overview.mdx

ChrisCostanzo-acme committed:
Update architecture-overview.mdx

ChrisCostanzo-acme committed:
More small docs refactors.

ChrisCostanzo-acme committed:
Fix mistype.

ChrisCostanzo-acme committed:
Documentation changes.

ChrisCostanzo-acme committed:
Point to git in pubspec again until publish issues resolved.

ChrisCostanzo-acme committed:
Update Domain and UI layer docs.

- Add more thorough explanation/justification for all of the different data containers.

ChrisCostanzo-acme committed:
Start on upgrading clean_framework_firestore to latest version of clean_framework.

ChrisCostanzo-acme committed:
Example project changes, add newest pubspec for each new package version.

ChrisCostanzo-acme committed:
Minor pubspec updates.

ChrisCostanzo-acme committed:
Update pubspecs release version publish.

ChrisCostanzo-acme committed:
Rename example project, version updates.

- Bump dependency versions.
- Renamed example for pub.dev

ChrisCostanzo-acme committed:
Fix clean_framework_firestore delete request bug.

- Updated pubspec and changelog in preparation for publish.
